### ISSUE TITLE
List all posts a group is subscribed to and/or include them in get_user_following_posts() if user is in groups

### ISSUE THREAD

**Issue by jeremyclarke** [Automattic/Edit-Flow#304]
I looked everwhere and searched for keywords that made sense but can't find an equivalent to get_user_following_posts() (notifications.php) that instead gets posts a group is following (i.e. get_group_following_posts())

The utility of such a function seems pretty obvious, but I need it specifically to soup up the "myposts" widget, which currently doesn't list posts the user is subscribed to because they are in a group (i.e. it only shows posts where their name is ticked in notifications, not their group). 

This means they are still addicted to email notifications for group subscriptions, which is annoying in many contexts, not least of which would be instances where the groups shift regularly. 

FWIW this problem (and many related ones) would also be satisfied if get_user_following_posts() also checked for relevant group subscriptions and included them. IMHO that's what most users would expect (my users did and were confused). 

From a quick analysis it seems the solution for that (souping up get_user_following_posts()) is to complicate it's tax_query to also query against $this->following_usergroups_taxonomy with terms for each group the user is in. Not a fast query but a logical one. 

Any thoughts on this? Would a patch with that update get committed, or should I work on this in an external silo?

**Comment by cojennin** [comment]
Makes sense to me. I would expect to see both posts that have my name ticked and posts that have the usergroup ticked in anything that dealt with showing me posts to which I'm subscribed. Unsure what kind of overhead it would add to the tax_query, but if you have some time to work on it that would be great! Can always figure out if it'll need improvement once we've got somewhere to start.

**Comment by cojennin** [comment]
@jeremyclarke any further thoughts on this? I would definitely be interested to see a patch for it!

Might be useful to make a function similar to `get_user_following_posts` called `get_usergroups_following_posts` as well as one that combines the functionality of the two (with the more complicated query you noted in your comment).

**Comment by GaryJones** [comment]
Closing as stale. This enhancement has been open for 9 years. The current following system works at the user level. If there's renewed interest in group-level following visibility, please open a new issue with specific use cases.
